sys/sys/types.h

```diff
@@ -251,6 +251,12 @@ struct cap_rights;
 typedef	struct cap_rights	cap_rights_t;
 #endif
 
+#ifdef __CHERI_PURE_CAPABILITY__
+typedef __uintptr_t	vm_ptr_t;
+#else
+typedef __vm_offset_t	vm_ptr_t;
+#endif
```

Comment:
I would still much prefer this to be called `vm_pointer_t` to match the fact that `vm_offset_t` spells "offset" in full. The type name continues to look very strange to me. It should be trivial to do a mass sed on the purecap kernel branch before extracting diffs.